cmd/loadOutputTranslation.go

```diff
@@ -0,0 +1,236 @@
+		} else if strings.Contains(line, clfsErrTag) {
+			c.processError(line)
+		} else {
+			//ignore all other output
+			//c.lcm.Info("ignored: " + line)
+		}
+	}
+
+	c.finishSignal <- ""
+}
+
+func (c *clfsExtensionOutputParser) finishParsing() {
+	<-c.finishSignal
+}
+
+func (c *clfsExtensionOutputParser) processInitialization(line string) {
+	cleanMessage := strings.TrimPrefix(line, clfsInitializationTag)
+
+	initInfo := clfsInitialization{}
+	err := json.Unmarshal([]byte(cleanMessage), &initInfo)
+	if err != nil {
+		// don't crash if these info are not critical and are only nice-to-have
+		return
+	}
+
+	c.lcm.Info(fmt.Sprintf("CLFSLoad Extension version: %s", initInfo.Version))
+	c.lcm.Info(fmt.Sprintf("CLFSLoad Extension configurations: compression type=%s, preserve hard links=%v",
+		initInfo.Compression, initInfo.PreserveHardLinks))
+
+	if initInfo.New {
+		c.lcm.Info("Starting a new job.")
+	}
+}
+
+func (c *clfsExtensionOutputParser) processInfo(line string) {
+	cleanMessage := strings.TrimPrefix(line, clfsInfoTag)
+
+	info := clfsInfo{}
+	err := json.Unmarshal([]byte(cleanMessage), &info)
+	if err != nil {
+		// don't crash if these info are not critical and are only nice-to-have
+		return
+	}
+
+	c.lcm.Info(fmt.Sprintf("%s phase has started.", info.Phase))
+}
+
+func (c *clfsExtensionOutputParser) processError(line string) {
+	parts := strings.Split(line, clfsErrTag)
+	if len(parts) < 2 {
+		return
+	}
+
+	c.lcm.Info(strings.TrimSpace(parts[1]))
+}
+
+func (c *clfsExtensionOutputParser) processProgress(line string) {
+	cleanMessage := strings.TrimPrefix(line, clfsProgressTag)
+
+	progress := clfsProgress{}
+	err := json.Unmarshal([]byte(cleanMessage), &progress)
+	if err != nil {
+		// abort, something went wrong
+		c.lcm.Error("Unable to handle the output from CLFS extension, please contact the dev team through Github.")
+	}
+
+	c.lcm.Progress(func(format common.OutputFormat) string {
+		if format == common.EOutputFormat.Text() {
+			return progress.parseIntoString()
+		} else if format == common.EOutputFormat.Json() {
+			// re-marshal it the Go way
+			jsonOutput, err := json.Marshal(progress)
+			common.PanicIfErr(err)
+			return string(jsonOutput)
+		}
+
+		return ""
+	})
+}
+
+func (c *clfsExtensionOutputParser) processEndOfJob(line string) {
+	cleanMessage := strings.TrimPrefix(line, clfsExitTag)
+
+	summary := clfsSummary{}
+	err := json.Unmarshal([]byte(cleanMessage), &summary)
+	if err != nil {
+		// abort, something went wrong
+		c.lcm.Error("Unable to handle the output from CLFS extension, please contact the dev team through Github.")
+	}
+
+	c.lcm.Exit(func(format common.OutputFormat) string {
+		if format == common.EOutputFormat.Text() {
+			return summary.parseIntoString()
+		} else if format == common.EOutputFormat.Json() {
+			// re-marshal it the Go way
+			jsonOutput, err := json.Marshal(summary)
+			common.PanicIfErr(err)
+			return string(jsonOutput)
+		}
+
+		return ""
+	}, common.EExitCode.Success())
```

Comment:
Is this always returning success? If so, isn't that inconsistent with what AzCopy normally does?  Don't we return failure if even one file failed?  Should we, and could we, follow the same pattern here?

Reply:
Sounds good I've fixed it.